### PR TITLE
Mark uncoverable statements and remove dead code

### PR DESCRIPTION
**Pull request by mimi1vx**
https://progress.opensuse.org/issues/121357

### CODE REVIEW

**Comment by okurz** [comment]
s/uncovereable/uncoverable/

File: t/18-qemu.t [hunk 1]
```diff
@@ -426,9 +426,9 @@ subtest 'qemu is not called on an empty file when ISO_1 is an empty string' => s
     my $mock_proc = Test::MockModule->new('OpenQA::Qemu::Proc');
     my $call_count = 0;
     $mock_proc->redefine(get_img_size => sub {
-            my ($iso) = @_;
-            $call_count++;
-            die 'get_img_size called on an empty string' unless $iso;
+            my ($iso) = @_;    # uncoverable statement
```
Comment (perlpunk) [inline t/18-qemu.t:429]: Hm, if a mocked function is never called, maybe it doesn't need to be mocked at all.
I just ran the test without this `redefine`, and it passed.

Comment (perlpunk) [reply t/18-qemu.t:429]: Funnily, this was added by me, so it seems I overlooked something.

Comment (mimi1vx) [reply t/18-qemu.t:429]: I think this function is called -> if tests fail :D

**Review by okurz — APPROVED** [review]

**Review by kalikiana — APPROVED** [review]

**Pull request merged**